### PR TITLE
#13 basic/extended test workflows from omc3.

### LINKED ISSUE #13: Bring back extended testing
:(

### PR DESCRIPTION
**Pull request by JoschD**
Fixes #13

### CODE REVIEW

**Comment by fsoubelet** [comment]
Why do we need this when we don't have tests yet? Do you already have plans for extended tests? As I see it, `optics_functions` should not have anything that takes a long time to run.

File: .github/workflows/basic_tests.yml [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
 
 jobs:
   tests:
     name: ${{ matrix.os }} / ${{ matrix.python-version }}
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04, macos-latest, windows-latest]
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        os: [ubuntu-latest, macos-latest, windows-latest]
```
Comment (fsoubelet) [inline .github/workflows/basic_tests.yml:19]: This removes 2 ubuntu versions and 1 Python version. Is this intentional?

Comment (JoschD) [reply .github/workflows/basic_tests.yml:19]: yes. I put all of that into extended tests

**Comment by JoschD** [comment]
> Why do we need this when we don't have tests yet? Do you already have plans for extended tests? As I see it, `optics_functions` should not have anything that takes a long time to run.

Yes. See my branch. 
I also put some tfs files in there, and one extended test is using them for example, while the basic test are running on "fake" dataframes

File: .github/workflows/basic_tests.yml (hunk 1 above)
Comment (fsoubelet) [reply .github/workflows/basic_tests.yml:19]: The basic tests should also test all python version and OSes

Comment (JoschD) [reply .github/workflows/basic_tests.yml:19]: Why? It's being tested before merging with all python versions and OSes anyway. 
I see it more like: 
Basic Tests: Did I break something obvious?
Extended Tests: Did I break something more advanced?

**Comment by fsoubelet** [comment]
Integration of these tbd later on.

**Pull request closed without merging**